presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -119,6 +119,29 @@ public static Slice parseLine(@SqlType(StandardTypes.VARCHAR) Slice input)
         return serialize(geometry);
     }
 
+    @Description("Returns a LineString from an array of points created with ST_Point")
+    @ScalarFunction("ST_LineString")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stLineString(@SqlType("array(" + GEOMETRY_TYPE_NAME + ")") Block input) {
+        if (input.getPositionCount() == 1) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "LineString cannot have one point");
+        }
+
+        MultiPath multipath = new Polyline();
+        for (int i = 0; i < input.getPositionCount(); i++) {
+            Slice slice = input.getSlice(i, 0, input.getSliceLength(i));
+            validateType("ST_Point", deserialize(slice), EnumSet.of(POINT));
+            OGCPoint point = (OGCPoint) deserialize(slice);
```

Comment:
`deserialize(slice)` is expensive. Call it once and store the results in a variable.